### PR TITLE
feat: process all ClassLoaders

### PR DESCRIPTION
**Pull request by szepeviktor**
- [ ] Added or updated tests
- [ ] Documented user facing changes
- [ ] Updated CHANGELOG.md

**Changes**

@canvural Please take a look at my sub-noob work. Thanks.

### CODE REVIEW

File: src/ApplicationResolver.php [hunk 1]
```diff
@@ -36,41 +36,36 @@ public static function resolve(): Application
...
-    }
-
-    protected static function getVendorDir(): ?string
-    {
-        $reflector = new ReflectionClass(ClassLoader::class);
-        $classLoaderPath = $reflector->getFileName();
-        if ($classLoaderPath === false) {
-            return null;
-        }
+        foreach (array_keys($classLoaders) as $vendorDir) {
```
Comment (szepeviktor) [inline src/ApplicationResolver.php:45]: Do we need `array_unique` here?

File: src/ApplicationResolver.php [hunk 2]
```diff
@@ -36,41 +36,36 @@ public static function resolve(): Application
...
+            }
 
-        $vendorDir = dirname($classLoaderPath, 2);
-        if (! is_dir($vendorDir)) {
-            return null;
+            $composerConfigPath = dirname($vendorDir).DIRECTORY_SEPARATOR.'composer.json';
+
+            if (file_exists($composerConfigPath)) {
+                self::$composer = json_decode((string) file_get_contents($composerConfigPath), true);
+                $namespace = (string) key(self::$composer['autoload']['psr-4']);
```
[the remark this answers is not in the document]
Comment (szepeviktor) [reply src/ApplicationResolver.php:54]: After being through this one, the next PR will do just that.

File: src/ApplicationResolver.php [hunk 3]
```diff
@@ -36,41 +36,36 @@ public static function resolve(): Application
...
-    {
-        $reflector = new ReflectionClass(ClassLoader::class);
-        $classLoaderPath = $reflector->getFileName();
-        if ($classLoaderPath === false) {
-            return null;
-        }
+        foreach (array_keys($classLoaders) as $vendorDir) {
+            if (substr($vendorDir, 0, 5) === 'phar:') {
+                continue;
+            }
```
Comment (canvural) [inline src/ApplicationResolver.php:48]: What about just doing `$vendorDir = dirname($app->basePath(), 3)`?

Comment (szepeviktor) [reply src/ApplicationResolver.php:48]: I'm just a janitor here.

`basePath` does not support custom loaders.

Adding a Laravel package to tests may help.

Comment (szepeviktor) [reply src/ApplicationResolver.php:48]: For example there is a Composer plugin called `bin-dir`

**Comment by canvural** [comment]
Closing because it is fixed in another way with #1392

**Pull request closed without merging**